svgdigitizer/svgplot.py

```diff
@@ -204,22 +220,182 @@ class SVGPlot:
     def __init__(
         self,
         svg,
-        xlabel=None,
-        ylabel=None,
+        page_orientation=PageOrientation.PORTRAIT,
         sampling_interval=None,
         curve=None,
         algorithm="axis-aligned",
     ):
         self.svg = svg
-        self.xlabel = xlabel or "x"
-        self.ylabel = ylabel or "y"
+        self.page_orientation = page_orientation
         self.sampling_interval = sampling_interval
         self._curve = curve
         self._algorithm = algorithm
 
     @property
     @cache
-    def axis_labels(self):
+    def xlabel(self):
+        r"""
+        Return the label of the x axis.
+
+        EXAMPLES::
+
+            >>> from svgdigitizer.svg import SVG
+            >>> from io import StringIO
+            >>> svg = SVG(StringIO(r'''
+            ... <svg>
+            ...   <g>
+            ...     <path d="M 0 200 L 0 100" />
+            ...     <text x="0" y="200">E1: 0 cm</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M 100 200 L 100 100" />
+            ...     <text x="100" y="200">E2: 1cm</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 100 L 0 100" />
+            ...     <text x="-100" y="100">j1: 0</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 0 L 0 0" />
+            ...     <text x="-100" y="0">j2: 1 A</text>
+            ...   </g>
+            ... </svg>'''))
+            >>> plot = SVGPlot(svg)
+            >>> plot.xlabel
+            'E'
+
+            >>> plot = SVGPlot(svg, page_orientation=PageOrientation.LANDSCAPE)
+            >>> plot.xlabel
+            'j'
+
+        """
+
+        for axis, orientation in self.axis_orientations.items():
+            if orientation == AxisOrientation.HORIZONTAL:
+                return axis
```

Comment:
```suggestion
        [axis] = [axis for axis in self.axis_orientations if self.axis_orientations[axis] == AxisOrientation.HORIZONTAL]
        return axis
```

I think this is shorter and has better error handling (reports an error if there is no such axis or more than one.)

Reply:
Very nice.